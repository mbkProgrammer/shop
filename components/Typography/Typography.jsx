/* eslint-disable default-case */
import { useTheme } from '@emotion/react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { useState } from 'react';

const Typography = ({ variant, children, css }) => {
  const theme = useTheme();

  let Variant;
  switch (variant) {
    case 'body1':
      Variant = styled.p`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.body.fontWeight};
        font-size: ${theme.typography.body.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'body2':
      Variant = styled.p`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.body.fontWeight};
        font-size: ${theme.typography.body.fontSize};
        padding: 0;
        text-align: justify;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'h1':
      Variant = styled.h1`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.h1.fontWeight};
        font-size: ${theme.typography.h1.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'h2':
      Variant = styled.h2`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.h2.fontWeight};
        font-size: ${theme.typography.h2.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'h3':
      Variant = styled.h3`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.h3.fontWeight};
        font-size: ${theme.typography.h3.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'h4':
      Variant = styled.h4`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.h4.fontWeight};
        font-size: ${theme.typography.h4.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'h5':
      Variant = styled.h5`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.h5.fontWeight};
        font-size: ${theme.typography.h5.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    case 'h6':
      Variant = styled.h6`
        color: ${theme.colors.text};
        font-weight: ${theme.typography.h6.fontWeight};
        font-size: ${theme.typography.h6.fontSize};
        padding: 0;
        line-height: 1.3;
        ${css}
      `;
      break;
    default:
      break;
  }

  return <Variant>{children}</Variant>;
};

Typography.propTypes = {
  variant: PropTypes.oneOf([
    'body1',
    'body2',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
  ]),
};

Typography.defaultProps = {
  variant: 'Body1',
};

export default Typography;
