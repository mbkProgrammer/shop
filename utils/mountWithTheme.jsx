import { ThemeProvider } from '@emotion/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../configs/Store';
import theme from '../configs/theme';

const mountWithTheme = (children) => renderer.create(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </Provider>,
);

export default mountWithTheme;
