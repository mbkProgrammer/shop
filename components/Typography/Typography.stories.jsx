import React from 'react';
import Typography from './Typography';

export default { title: 'components/Typography' };
export const Body1 = () => (
  <Typography variant="body1">
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Harum maiores
    repellat, eos commodi, cupiditate possimus atque cum veniam ipsum natus vel
    vero a veritatis minima qui numquam. Suscipit incidunt necessitatibus nihil
    quos ab itaque voluptatum officia explicabo, deleniti sint nemo neque cum
    illo autem vero molestias recusandae id aliquam veritatis, laboriosam
    voluptatem. Dolorem fuga ullam sequi obcaecati maxime cupiditate molestiae?
  </Typography>
);
export const Body2 = () => (
  <Typography variant="body2">
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Harum maiores
    repellat, eos commodi, cupiditate possimus atque cum veniam ipsum natus vel
    vero a veritatis minima qui numquam. Suscipit incidunt necessitatibus nihil
    quos ab itaque voluptatum officia explicabo, deleniti sint nemo neque cum
    illo autem vero molestias recusandae id aliquam veritatis, laboriosam
    voluptatem. Dolorem fuga ullam sequi obcaecati maxime cupiditate molestiae?
  </Typography>
);
export const H1 = () => <Typography variant="h1">simple text</Typography>;
export const H2 = () => <Typography variant="h2">simple text</Typography>;
export const H3 = () => <Typography variant="h3">simple text</Typography>;
export const H4 = () => <Typography variant="h4">simple text</Typography>;
export const H5 = () => <Typography variant="h5">simple text</Typography>;
export const H6 = () => <Typography variant="h6">simple text</Typography>;
