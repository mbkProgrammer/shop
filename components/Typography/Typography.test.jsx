import mountWithTheme from '../../utils/mountWithTheme';
import Typography from './Typography';

test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="h1">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="h2">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="h3">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="h4">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="h5">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="h6">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="body1">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Typography smaple', () => {
  const component = mountWithTheme(
    <Typography variant="body2">Hi Mbk</Typography>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
