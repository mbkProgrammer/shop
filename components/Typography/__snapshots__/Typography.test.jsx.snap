// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography smaple 1`] = `
<h1
  className="css-xkr7dq"
>
  Hi Mbk
</h1>
`;

exports[`Typography smaple 2`] = `
<h2
  className="css-1484rop"
>
  Hi Mbk
</h2>
`;

exports[`Typography smaple 3`] = `
<h3
  className="css-1bj96s3"
>
  Hi Mbk
</h3>
`;

exports[`Typography smaple 4`] = `
<h4
  className="css-124arhk"
>
  Hi Mbk
</h4>
`;

exports[`Typography smaple 5`] = `
<h5
  className="css-b3i3zq"
>
  Hi Mbk
</h5>
`;

exports[`Typography smaple 6`] = `
<h6
  className="css-kyp4p2"
>
  Hi Mbk
</h6>
`;

exports[`Typography smaple 7`] = `
<p
  className="css-16v1ac2"
>
  Hi Mbk
</p>
`;

exports[`Typography smaple 8`] = `
<p
  className="css-v0dx9j"
>
  Hi Mbk
</p>
`;
